---
title: "Using KOIOS"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Using KOIOS}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{css, echo=FALSE}
body .main-container {
max-width: 1280px !important;
width: 1280px !important;
}
body {
max-width: 1280px !important;
}
```

```{r setup, include=FALSE}
library(KOIOS)

knitr::opts_chunk$set(echo = TRUE)

```

KOIOS provides users with a lightweight and easy tool that can generate a set of OMOP Concept IDs, and related information, from a user input VCF file, utilizing the ClinGen Allele Registry test network. This tool is intended to be used by users with no experience of VCF parsing, though some previous experience may help.

## The VCF

VCF files may contain a lot of varied information from different sources, but a [valid VCF](https://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40/) file must always contain 8 fixed fields per record. KOIOS relies on the following fixed fields to generate variant and allele data from an input VCF: Chromsome, Position, Reference Allele and Alternate Allele.

VCF data is also highly dependent on the reference genome from which it was generated. Currently KOIOS supports the following reference genomes: Hg18, Hg19 and Hg38. An automatic mode for detecting the reference used to generate a file is also available.

## Running KOIOS

KOIOS relies on an active internet conncetion and on the user having a VCF file that can be loaded in by the VCFr:: package.

We will run this vignette using data downloaded from the [GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE142442).

KOIOS utilizes the OMOP Genomic Vocabulary, derived from [ATHENA](https://athena.ohdsi.org/vocabulary/list).

```{r}

#Load concept library, derived from ATHENA
concepts <- loadConcepts()

#Load input VCF
vcf <- loadVCF(here::here("inst/extdata/example/GSE142442_27_AST.vcf"))

#Indicate whether we would like to generate information on all transcript alleles associated with a given variant.
generateTranscripts <- "TRUE"

```

### Reference Genome

To utilise KOIOS, the correct reference genome must be supplied or 

```{r}
#Set the correct reference genome - hg18, hg19, hg38
ref <- "hg19"
```

If a user is unaware of the reference genome that was used to generate a given file, they may use the findReference() function. The findReference function selects 10 single nucleotide variants from the given VCF and checks them against the reference of each genome at that position. We then compare which reference genome returned a full set of valid results and use this as the reference genome going forward.

```{r}

ref <- findReference(vcf = vcf)
print(ref)

```

### Running the ClinGen parser

Once the input and settings are established, KOIOS is run as follows:

```{r, fig.align='center'}

#Load reference
ref.df <- loadReference(ref)

#Process VCF file
vcf.df <- processVCF(vcf)

#Generate HGVSG formatted data for submission to ClinGen
vcf.df <- generateHGVSG(vcf = vcf.df, ref = ref.df)

#Generate URLs and download and process all relevant JSON data
alleles.df <- processClinGen(vcf.df,ref,generateAll = generateTranscripts, progressBar = F)

knitr::kable(alleles.df[c(1:8,97:105),], format = "html", align = "c")

```

The alleles.df dataframe contains all the data for each supplied variant that was found within the ClinGen regtest network. Information on the relevant transcript variants, protein variants and effected genes are all returned here and can be easily explored. Each allele # refers to an individual input variant. Each input genomic variant will likely be attached to several transcripts and proteins, as well as other genomic variants that are derived from other reference genomes than the input reference.

### Filtering via OMOP Genomic

Now that our allele data has been returned, we wish to perform the final step in the KOIOS process and link this directly to the genomic vocabulary. This can be done easily via the addConcepts function. Users can choose to return all alleles, or only those which were mapped to an existing concept ID.

```{r}

#Match returned variant data with the OMOP concept library
concepts.df <- addConcepts(alleles.df, concepts, returnAll = FALSE)

concepts.df <- concepts.df[!duplicated(concepts.df$hgvsg),]

knitr::kable(concepts.df[,c(3,2,5,6,7,8)], format = "html", align = "c")

```

This output data frame will contain all the relevant information a user would need to track a variant within the OMOP Genomic Vocabulary through their data.


